### PR TITLE
Feature/improve tests

### PR DESCRIPTION
**Pull request by binarymash**
Basically consolidates all the common middleware test server functionality into a common base class, which makes the tests a lot cleaner and consistent. Killed a few build warnings relating to the tests. Also adds a bit more test coverage, so I've upped the minimum coverage threshold.

### CODE REVIEW

**Comment by binarymash** [comment]
Not sure why the build is failing in AppVeyor... maybe an issue with Appveyor or Cake.

**Comment by binarymash** [comment]
Looks like a nuget issue on appveyor... https://appveyor.statuspage.io/

**Comment by TomPallister** [comment]
@binarymash thanks man, I'm on it!

File: test/Ocelot.UnitTests/Responder/HttpContextResponderTests.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            _removeOutputHeaders = new Mock<IRemoveOutputHeaders>();
+            _httpContext = new DefaultHttpContext();
+            _httpResponseMessage = new HttpResponseMessage();
+            _httpResponseMessage.Content = new MyHttpContent();
+            _responder = new HttpContextResponder(_removeOutputHeaders.Object);
+        }
+
+        [Fact]
+        public void DoSomething()
```
Comment (TomPallister) [inline test/Ocelot.UnitTests/Responder/HttpContextResponderTests.cs]: Is this the name you intended to leave? Almost feels appropriate tbh!

File: test/Ocelot.UnitTests/ServerHostedMiddlewareTest.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    using System.IO;
+    using System.Net.Http;
+    using Microsoft.AspNetCore.TestHost;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.AspNetCore.Builder;
+    using Moq;
+    using Ocelot.Infrastructure.RequestData;
+
+    public abstract class ServerHostedMiddlewareTest : IDisposable
```
Comment (TomPallister) [inline test/Ocelot.UnitTests/ServerHostedMiddlewareTest.cs]: This is a way better approach 😄

File: test/Ocelot.UnitTests/Responder/HttpContextResponderTests.cs (hunk 1 above)
Comment (binarymash) [inline test/Ocelot.UnitTests/Responder/HttpContextResponderTests.cs]: oh I should have deleted that file - this is what I started writing before I realised why you're using the test server. I'll remove it.

**Comment by binarymash** [comment]
Lets wait a wee bit before merging, see if appveyor fix things.

**Comment by TomPallister** [comment]
@binarymash building now! Thanks for the help!!!
